### PR TITLE
country-specific set-aside shares

### PR DESCRIPTION
**Pull request by flohump**
## :bird: Purpose of this PR :bird:

- output/extra/disaggregation.R updated to account for country-specific set-aside shares in post-processing

## :wrench: Checklist for PR creator :wrench:

- [x] Labeling pull request correctly [from the label list](https://github.com/magpiemodel/magpie/labels).

* Low risk : Simple bugfixes (missing files, updated documentation, typos) or Start/output scripts
* Medium risk : New realization / Changes to existing realization / Other changes which don't modify default.cfg
* High risk : New input files (if cfg$input is changed in default.cfg) / Modification to core model (eg. changes in equations, calculations, introduction of new sets etc.) / Other changes in default.cfg

- [x] Providing additional information based on PR label

* Low risk : No new model run needed.
* Medium risk : Default run based on the current version of the fork from which PR is made
* High risk
  * Default run from the current develop branch
  * Default run based on the current version of the fork from which PR is made

- [x] :chart_with_downwards_trend: Performance loss/gain from current default behavior :chart_with_upwards_trend:
  * Current develop branch default : ** mins
  * This PR's default :  ** mins

- [x] Added changes to `CHANGELOG.md`
- [x] Compilation check (model starts without compilation errors - use `gams main.gms action=c` in model folder for testing).
- [x] No hard coded numbers and cluster/country/region names.
- [x] The new code doesn't contain declared but unused parameters or variables.
- [x] Where relevant, In-code comments added including documentation comments.
- [x] Made sure that documentation created with [`goxygen`](https://github.com/pik-piam/goxygen) is okay (use `goxygen::goxygen()` for testing).
- [x] Changes to [`magpie4`](https://github.com/pik-piam/magpie4) R library for post processing of model output (ideally backward compatible).
- [x] Self-review of my own code.
- [x] For high risk runs: validation of major model indicators - Land-use, emissions, food prices, Tau.  %Delete this line in case it is not a high risk run%

## :warning: Additional notes or warnings :warning:
NA

## :rotating_light: Checklist for RSE reviewer :rotating_light:

- [x] PR is labeled correctly.
- [x] `CHANGELOG` is updated correctly
- [x] No hard coded numbers and cluster/country/region names.
- [x] No unnecessary increase in module interfaces
- [ ] All required updates in interfaces (if any) have been properly adressed in the module contracts
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

## :rotating_light: Checklist for MAgPIE reviewer :rotating_light:

- [x] PR is labeled correctly.
- [x] `CHANGELOG` is updated correctly
- [x] No hard coded numbers and cluster/country/region names.
- [x] Changes to the model are scientifically sound
- [x] In-code comments and documentation comments are satisfactory.
- [x] model behavior/performance is satisfactory.
- [x] Requested changes (if any) were applied correctly

### CODE REVIEW

File: scripts/output/extra/disaggregation.R [hunk 1]
```diff
@@ -49,6 +49,14 @@ if (cfg$gms$crop=="endo_apr21"){
...
+  iso <- readGDX(gdx, "iso")
+  set_aside_iso <- readGDX(gdx,"policy_countries30")
+  set_aside_select <- readGDX(gdx, "s30_set_aside_shr")
+  set_aside_noselect <- readGDX(gdx, "s30_set_aside_shr_noselect")
+  set_aside_shr <- new.magpie(iso, fill = set_aside_noselect)
+  set_aside_shr[set_aside_iso,,] <- set_aside_select
+  
   avl_cropland_hr <- file.path(outputdir, "avl_cropland_0.5.mz")       # available cropland (at high resolution)
   marginal_land <- cfg$gms$c30_marginal_land                      # marginal land scenario
   set_aside_shr <- cfg$gms$s30_set_aside_shr                      # set aside share (default: 0)
```
Comment (tscheypidi) [inline scripts/output/extra/disaggregation.R:62]: It looks to me like if this line would overwrite all your changes again. Please check

**Review by tscheypidi — CHANGES_REQUESTED** [review]
I believe your changes are overwritten again a few lines later. Please check.

File: scripts/output/extra/disaggregation.R (hunk 1 above)
Comment (flohump) [reply scripts/output/extra/disaggregation.R]: Thanks for spotting this. @pvjeetze I deleted line 62.

**Review by tscheypidi — APPROVED** [review]

**Review by pvjeetze — APPROVED** [review]

File: scripts/output/extra/disaggregation.R (hunk 1 above)
Comment (pvjeetze) [reply scripts/output/extra/disaggregation.R]: A nice to have, maybe: A brief comment line, what the new lines are doing..

Comment (flohump) [reply scripts/output/extra/disaggregation.R]: I added a comment

**Pull request merged**